agent_api_rest/src/verification/relying_party/redirect.rs

```diff
@@ -0,0 +1,184 @@
+    use agent_store::{in_memory, EventPublisher};
+    use agent_verification::services::test_utils::test_verification_services;
+    use axum::{
+        body::Body,
+        http::{self, Request},
+        Router,
+    };
+    use oid4vc_core::{
+        authorization_request::{AuthorizationRequest, Object},
+        client_metadata::ClientMetadataResource,
+        scope::Scope,
+        DidMethod, SubjectSyntaxType,
+    };
+    use oid4vc_manager::ProviderManager;
+    use siopv2::authorization_request::ClientMetadataParameters;
+    use tower::Service;
+    use wiremock::{
+        matchers::{method, path},
+        Mock, MockServer, ResponseTemplate,
+    };
+
+    pub async fn redirect(app: &mut Router, state: String) {
+        let authorization_request = AuthorizationRequest::<Object<SIOPv2>>::builder()
+            .client_id("client_id".to_string())
+            .scope(Scope::openid())
+            .redirect_uri("https://example.com".parse::<url::Url>().unwrap())
+            .response_mode("direct_post".to_string())
+            .client_metadata(ClientMetadataResource::ClientMetadata {
+                client_name: None,
+                logo_uri: None,
+                extension: ClientMetadataParameters {
+                    subject_syntax_types_supported: vec![SubjectSyntaxType::Did(
+                        DidMethod::from_str("did:key").unwrap(),
+                    )],
+                },
+            })
+            .nonce("nonce".to_string())
+            .state(state)
+            .build()
+            .unwrap();
+
+        let provider_manager =
+            ProviderManager::new([Arc::new(futures::executor::block_on(async { secret_manager().await }))]).unwrap();
+        let authorization_response = provider_manager
+            .generate_response(&authorization_request, Default::default())
+            .unwrap();
+
+        let response = app
+            .call(
+                Request::builder()
+                    .method(http::Method::POST)
+                    .uri("/redirect")
+                    .header(
+                        http::header::CONTENT_TYPE,
+                        mime::APPLICATION_WWW_FORM_URLENCODED.as_ref(),
+                    )
+                    .body(Body::from(serde_urlencoded::to_string(authorization_response).unwrap()))
+                    .unwrap(),
+            )
+            .await
+            .unwrap();
+
+        assert_eq!(response.status(), StatusCode::OK);
+    }
+
+    #[tokio::test]
+    #[tracing_test::traced_test]
+    async fn test_redirect_endpoint() {
+        let mock_server = MockServer::start().await;
+
+        Mock::given(method("POST"))
+            .and(path("/ssi-events-subscriber"))
+            .respond_with(ResponseTemplate::new(200))
+            .mount(&mock_server)
+            .await;
+
+        let target_url = format!("{}/ssi-events-subscriber", &mock_server.uri());
+
+        TEST_EVENT_PUBLISHER_HTTP_CONFIG.lock().unwrap().replace(
```

Comment:
Why is the "redirect" test also testing the "event publisher"?

Reply:
I just wanted an extra assertion that the events are indeed successfully dispatched by the publisher in the complete
SIOPv2 flow. I do want to keep a test like this but I also contemplated whether this is the correct place to do it. The
best solution is probably to soon start writing a bunch of integration tests in `agent_application` where this test can
be migrated to as well.